RadixWallet/Features/DappsAndPersonas/DappDetails/DappDetails.swift

```diff
@@ -184,13 +189,24 @@ public struct DappDetails: Sendable, FeatureReducer {
 			state.$resources = .loading
 			let dAppID = state.dAppDefinitionAddress
 			let checkForAuthorizedDapp = state.authorizedDapp == nil
-			return .run { send in
+			return .run { [dAppDefinitionAddress = state.dAppDefinitionAddress] send in
 				if checkForAuthorizedDapp, let authorizedDapp = try? await authorizedDappsClient.getDetailedDapp(dAppID) {
 					await send(.internal(.dAppUpdated(authorizedDapp)))
 				}
 
 				let result = await TaskResult {
-					try await onLedgerEntitiesClient.getAssociatedDapp(dAppID).metadata
+					let metadata = try await onLedgerEntitiesClient.getAssociatedDapp(dAppID).metadata
+
+					if let url = metadata.claimedWebsites?.first {
+						do {
+							try await rolaClient.performWellKnownFileCheck(url, dAppDefinitionAddress)
+							await send(.internal(.mainWebsiteValidated(url)))
+						} catch {
+							loggerGlobal.error("Failed to validate dapp main website: \(error)")
+						}
+					}
```

Comment:
I would rather do this in the `metadataLoaded` action run block, to be consistent with how other info is extracted from the metadata.